pagopa/io-app
# feat: [IOAPPCOM-1] Add preview flag on notification onboarding

# ⚠️ This PR depends on #4139 ⚠️
Please review first #4139 in order to avoid duplicate review work.

## Short description
This PR completes the Push Notification Opt-In screen during the on-boarding procedure.

![Screenshot 2022-11-02 at 14 28 53](https://user-images.githubusercontent.com/5150343/199502074-f207c06b-5660-4200-82e3-36f416784dc8.png)

## List of changes proposed in this pull request
- Push Notification Opt-In screen has the "Show previews" toggle above the "Allow reminders" toggle
- Example Notification view (in the Push Notification Opt-In screen) changes its title and subtitle reflecting the toggles combinations
- Network request (upon bottom page Continue button pressing) sends the "Show Preview" value to the server (along with the "Allow reminders" value)
- Opt-In notification saga checks for both "reminder_status" and "push_notifications_content_type" value definitions in order to determine whether to skip or not the showing of the Opt-In Notification view, during the on-boarding
- Bottom Sheet component has been reused between this view and the Notification Preferences view, on the profile notification settings, in order to display the same content (for the "Show Preview" description)

## How to test
In order to display this screen, you have to run an io-dev-api-server instance where the profile.firstOnboarding flag has been set to "true" once and to "false", upon restarting of both application and dev-server instances.
With such dev-server flag set to "true", start the application and go through the on-boarding process. Make sure to test that:
- With the server flag set to "true", the Opt-in screen is shown after having confirmed the email address;
- Labels and texts are correct
- Both toggles are on
- Changing a toggle value does not trigger a network request
- The example notification content matches the 4 states <title, subtitle> that are shown in the [FIGMA file](https://www.figma.com/file/nWCT5aGx51j2rVz0Xa95mr/Opt-in-notifiche?node-id=0%3A1), depending on the toggles values
- The "More Info" link is working, displaying the requested information on the bottom sheet
- The Bottom sheet is closable after having been opened
- The Contextual Help is openable from the top right "Help" button
- Toggles values are sent to the server after pressing the "Continue" button
- The next shown screen is the "Service configuration" one (during the on boarding with the server flag set to "true")
- Selected Toggle values are reflected on the "profile tab", under "Preferences" -> "Push notifications"

- With the server flag set to "false", the Opt-in screen is shown after having unlocked the application with the previously registered pin
- With the server flag set to "false" and having already sent the toggles values to the server (using the "continue" button on this new screen), the Opt-In Notification Screen is not displayed anymore

- The "More info" link in the "Show Preview" Toggle description is still working, on the "Profile" tab, under "Preferences" -> "Push notifications"
